maven-plugin/plugin-core/src/main/java/org/jvnet/jaxb/maven/util/JarScanner.java

```diff
@@ -52,9 +54,9 @@ public void scan() {
             while (jarFileEntries.hasMoreElements()) {
                 JarEntry entry = jarFileEntries.nextElement();
                 String name = entry.getName();
-                if (name.startsWith("..") || name.startsWith("/")) {
-                    // ignore "zip slip" file pattern attack
-                    continue;
+                File file = new File(destinationDir, entry.getName());
+                if (!file.toPath().normalize().startsWith(destinationDir.toPath())) {
```

Comment:
Should we look for contains '..' ?

Reply:
By normalizing the path, this will check that final path is not outside the `destinationDir` (if contains .. and makes it go outside the `destinationDir`, file will be refused)